COM-588 Implement LTFU report

tmvumbi2 opened the pull request:
Implements LTFU report
Jira ticket: https://jembiprojects.jira.com/browse/COM-588

tmvumbi2 committed:
COM-588 Implement LTFU report
Implements LTFU report

lauravignoli commented on openmrs/apps/reports/HIV_PATIENTS_LOST_TO_FOLLOWUP/sql/dissagregated_by_patient.sql:
```diff
@@ -1,16 +1,24 @@
+    patientHasEnrolledIntoHivProgram(patient_id) = "Yes" AND
+    patientHasTherapeuticLine(patient_id, 0) AND
+    patientHasStartedARVTreatmentBefore(patient_id, '#endDate#') AND
+    patientEndOfARVTreatmentIsBefore(patient_id, '#startDate#') AND
```
@tmvumbi2 Why check that the ARV treatment ended before the start of the reporting period? The spec says: **Drug must not end before the start of the reporting quarter**

The existing function **patientWasOnARVTreatmentAtEndReportingPeriod** could be used here?

lauravignoli replied:
Maybe we could rename patientWasOnARVTreatmentAtEndReportingPeriod to patientWasOnARVTreatmentByDate or something like this?

tmvumbi2 committed:
COM-588 Refactor: re-use existing function
Renames patientWasOnARVTreatmentAtEndReportingPeriod to patientWasOnARVTreatmentByDate (more generic) and uses it in LTFU report
